### PR TITLE
Add test for driver-only RSA (crypto only)

### PR DESCRIPTION
**Pull request by lpy4105**
## Description

Fix: #8553 

Two test components added:
- `test_psa_crypto_config_accel_rsa_crypto`
- `test_psa_crypto_config_reference_rsa_crypto`

## PR checklist

- [x] **changelog** not required, test components
- [x] **backport** not required, no driver support on `2.28`
- [x] **tests** not required

### CODE REVIEW

File: tests/scripts/all.sh
Comment (lpy4105) [inline tests/scripts/all.sh:3456]: Could we think `component_test_psa_crypto_config_accel_rsa_signature` is a duplication？
It starts from `default` config but disables many RSA features in X.509 and TLS.

**Review by lpy4105 — COMMENTED** [review]
A question for reviewers.

File: tests/scripts/all.sh
Comment (mpg) [inline tests/scripts/all.sh:3486]: Minor: this could be simplified to `not grep mbedtls_rsa library/rsa.o` I think.

File: tests/suites/test_suite_psa_crypto.function
Comment (mpg) [inline tests/suites/test_suite_psa_crypto.function:9690]: Question: it's not obvious to me why you picked those exact guards, can you explain?

File: tests/scripts/analyze_outcomes.py
Comment (mpg) [inline tests/scripts/analyze_outcomes.py:513]: Typo: stuff (I guess)

File: tests/scripts/analyze_outcomes.py
Comment (mpg) [inline tests/scripts/analyze_outcomes.py:520]: Thanks for the clear comments (here and below), that's very helpful!

**Review by mpg — COMMENTED** [review]
Looks pretty good to me, thanks! Just minor things, plus one question about something I need to understand better before I can approve.

File: tests/suites/test_suite_psa_crypto.function
Comment (lpy4105) [reply tests/suites/test_suite_psa_crypto.function:9690]: These macros come from:
https://github.com/Mbed-TLS/mbedtls/blob/f3ccfddb45dc964124145789e433f22d382ca5a6/library/psa_crypto.c#L1375-L1405

If these conditions are not fitted, `psa_get_key_attributes` can't get RSA key attributes but returns `PSA_SUCCESS`.

Comment (mpg) [reply tests/suites/test_suite_psa_crypto.function:9690]: Thanks for the explanation! I'll note the comment there is outdated (references an issue that was closed) and I'm tempted to slightly expand the scope of that PR to include updating that comment, but I'm not sure what the proper replacement would be yet.

@gilles-peskine-arm It looks like the prediction in [the last comment in the issue listed above](https://github.com/ARMmbed/mbed-crypto/issues/216#issuecomment-1561795353) was not realised, as we seem to now find ourselves in a situation where we have an (experimental, in `crypto_extra.h`) API for custom domains parameters, with `psa_set/get_key_domain_parameters()` allowing to set/get custom public exponents for RSA keys, but no corresponding support in the driver API that I could find.

More precisely, I'm not sure how `psa_get_key_attributes()` is supposed to populate the attributes structure with the public exponent in case RSA keys are supported only by a driver? I think it's OK if we have a gap for now (which could be closed either by expanding driver support or by removing the experimental `domain_parameters` extension), but then we need to make sure we have an open issue tracking it.

Also, if we accept this gap for now, I'm a bit uncomfortable with the fact that `psa_get_key_attributes()` and then `psa_get_key_domain_parameters()` will both return success even when the actual value of `e` can't be accessed. I'd say this would deserve at least a warning in the documentation of `psa_get_key_domain_parameters()` that with RSA keys handled by a driver, the exponent will always be reported as the default one regardless of its actual value. Or should we make `psa_get_key_domain_parameters()` return an error in this case?

**Review by mpg — COMMENTED** [review]
LGTM, thanks!

I'm withholding approval for now because I still want to understand better the limitation you uncovered about custom domain parameters (ie, non-default public exponents). It probably won't be resolved here, but I'd like to make sure we have task(s) created to track it before closing that discussion.

File: tests/suites/test_suite_psa_crypto.function
Comment (gilles-peskine-arm) [reply tests/suites/test_suite_psa_crypto.function:9690]: We're in the uncomfortable position where domain parameters are deprecated, so we haven't included them in newer parts of the library, such as the driver interface. But we haven't removed them partly because removing takes time and partly because there is one thing that we care about that requires them, namely generating an RSA key with a non-default exponent.

My preferred resolution would be to:

1. Finish [designing `psa_generate_key_ext`](https://github.com/Mbed-TLS/mbedtls/issues/6494) and implement it and its driver counterpart.
2. In a subsequent minor release, remove domain parameters.
3. Now you don't have this problem anymore.

But obviously that would take time that we don't have.

In the meantime, I think ideally `psa_get_key_domain_parameters` should return an error if it can't obtain the public exponent. `psa_get_key_attributes` must always succeed (except if access to the key object fails in some way, e.g. an invalid key identifier or an operational error).

**Review by mpg — APPROVED** [review]
LGTM if the CI agrees

**Review by valeriosetti — APPROVED** [review]
LGTM

File: library/psa_crypto.c
Comment (gilles-peskine-arm) [inline library/psa_crypto.c:1407]: Setting `domain_parameters_size` to `SIZE_MAX` is wrong: it's a promise that you can read `SIZE_MAX` bytes from `domain_parameters`. This might not be critical since this is a private field, and its only accessor function `psa_get_domain_parameters` takes care of that case. But at the very least, it needs to be documented in the definition of `struct psa_key_attributes_s` in `psa/crypto_struct.h`.

https://github.com/Mbed-TLS/mbedtls/pull/8644

**Pull request merged**
